src/models/metadata-fields/field-types/number.ts

```diff
@@ -0,0 +1,28 @@
+import {
+  FieldParserInterface,
+  FieldParserRawValue,
+} from '../field-parser-interface';
+import { MetadataField, MetadataRawValue } from '../metadata-field';
+
+export class NumberParser implements FieldParserInterface<number> {
+  // use a shared static instance for performance instead of
+  // instantiating a new instance for every use
+  static shared = new NumberParser();
+
+  parseValue(rawValue: FieldParserRawValue): number | undefined {
+    if (typeof rawValue === 'number') return rawValue;
+    if (typeof rawValue === 'boolean') return undefined;
```

Comment:
This is mostly for protection. `FieldParserRawValue` can be a `number`, `string`, or a `boolean` (the types we can get back from the search engine/metadata response), and since we're in a `Number` field, if we get a boolean back, we can't convert that to a number.